packages/flutter/test/material/dropdown_theme_test.dart

```diff
@@ -0,0 +1,633 @@
+      ),
+    ));
+
+    WidgetsBinding.instance.focusManager.highlightStrategy = FocusHighlightStrategy.alwaysTraditional;
+    focusNode.requestFocus();
+    await tester.pumpAndSettle();
+    // Test `focusColor`.
+    final RenderObject inkFeatures = tester.allRenderObjects.firstWhere((RenderObject object) => object.runtimeType.toString() == '_RenderInkFeatures');
+    expect(inkFeatures, paints..rect(color: const Color(0xff00ff00)));
+
+    // Test `style`.
+    Color textColor(String text) {
+      return tester.renderObject<RenderParagraph>(find.text(text)).text.style!.color!;
+    }
+    expect(textColor('two'), Colors.red);
+
+    // Test `iconColor`.
+    final RichText enabledIconRichText = tester.widget<RichText>(_iconRichText(iconKey));
+    expect(enabledIconRichText.text.style!.color, Colors.blue);
+
+    await tester.tap(find.text(value));
+    await tester.pumpAndSettle();
+
+    // Test `menuMaxHeight` and `borderRadius`.
+    expect(
+      find.ancestor(
+        of: find.text(value).last,
+        matching: find.byType(CustomPaint),
+      ).at(2),
+      paints
+        ..save()
+        ..rrect()
+        ..rrect()
+        ..rrect()
+        ..rrect(rrect: const RRect.fromLTRBXY(0.0, 0.0, 134.0, 180.0, 18.0, 18.0)),
+    );
+
+    // Test `dropdownColor`.
+    expect(
+      find.ancestor(
+        of: find.text(value).last,
+        matching: find.byType(CustomPaint),
+      ).at(2),
+      paints
+        ..save()
+        ..rrect()
+        ..rrect()
+        ..rrect()
+        ..rrect(color: const Color(0xff0000ff), hasMaskFilter: false),
+    );
+
+    await tester.pumpWidget(MaterialApp(
+      theme: ThemeData(dropdownTheme: _dropdownTheme()),
+      home: Scaffold(
+        body: Center(
+          child: buildDropdown(
+            isFormField: true,
+            icon: customIcon,
+            autofocus: true,
+            focusNode: focusNode,
+            onChanged: onChanged,
+          ),
+        ),
+      ),
+    ));
+    await tester.pumpAndSettle();
+
+    final InputDecoration inputDecoration = tester.widget<InputDecorator>(
+      find.byType(InputDecorator)
+    ).decoration;
+    expect(inputDecoration.floatingLabelAlignment, FloatingLabelAlignment.center);
+
+    await tester.pumpWidget(MaterialApp(
+      theme: ThemeData(dropdownTheme: _dropdownTheme()),
+      home: Scaffold(
+        body: Center(
+          child: buildDropdown(
+            isFormField: false,
+            icon: customIcon,
+            autofocus: true,
+            focusNode: focusNode,
+            items: null,
+            onChanged: onChanged,
+          ),
+        ),
+      ),
+    ));
+    await tester.pumpAndSettle();
+
+    // Test disabled `iconColor`.
+    final RichText disabledIconRichText = tester.widget<RichText>(_iconRichText(iconKey));
+    expect(disabledIconRichText.text.style!.color, Colors.black26);
+  });
+
+  testWidgets('Dropdown uses local theme over global theme', (WidgetTester tester) async {
```

Comment:
Can you also add a test that verifies if I set `Dropdown.<property>` it is used instead of `DropdownThemeData.<property>`?
Your buildDropdown method is already set up to pass the parameters, although right now many of them are not used. If they aren't used in tests they can be removed from the boilerplate.

Reply:
> Can you also add a test that verifies if I set Dropdown.<property> it is used instead of DropdownThemeData.<property>?

Added the test.

> Your buildDropdown method is already set up to pass the parameters, although right now many of them are not used. If they aren't used in tests they can be removed from the boilerplate.

Removed unused parameters.

Thank you!